src/main/java/seedu/address/logic/commands/ImportCommand.java

```diff
@@ -68,17 +81,68 @@ public ImportCommand(Path filePath) {
         this.filePath = filePath;
     }
 
+    /**
+     * Generates a report of the import process.
+     */
+    private String generateReport() {
+        String importSuccessMsg = (
+                !errorMsgsFromReadingCsv.isEmpty() | !errorMsgsFromAddingPersons.isEmpty()
+                        ? "Import completed with errors\n"
+                        : successfulImports == 0
+                        ? "No persons were imported\n"
+                        : MESSAGE_IMPORT_SUCCESS);
+
+        String reportForAddingPersons = "\n" + (
+                errorMsgsFromAddingPersons.isEmpty() && successfulImports > 0
+                        ? "All valid persons have been added!\n"
+                        : successfulImports == 0
+                        ? "No valid persons were found. Csv file is empty or error occurred reading from csv file\n"
+                        : "Errors found in adding persons!\n")
+            + String.format("Successful imports: %d\n", successfulImports)
+            + String.format("Unsuccessful imports: %d\n", unsuccessfulImports)
+            + errorMsgsFromAddingPersons;
+
+        String reportForReadingCsv =
+                errorMsgsFromReadingCsv.isEmpty()
+                        ? ""
+                        : "\nErrors found from reading csv!\n" + errorMsgsFromReadingCsv;
+
+        return importSuccessMsg + reportForReadingCsv + reportForAddingPersons;
+    }
+    /**
+     * Generates an error report from adding persons. The index refers to the index of the person in personsData.
+     * @param e
+     * @param index
+     */
+    private void generateErrorReportFromAddingPersons(Exception e, int index) {
+        errorMsgsFromAddingPersons += String.format("Person %s: ", index) + e.getMessage() + "\n";
+    }
+
+    private void generateErrorReportFromReadingCsv(DataLoadingException e) {
+        errorMsgsFromReadingCsv += e.getMessage() + "\n";
+    }
+
     @Override
     public CommandResult execute(Model model) throws CommandException {
```

Comment:
Short execute method that has most of the complicated functionality abstracted away. Become the complicated return types for the data and error report, I think this is very well done.